packages/cli/src/cmds/beacon/initBeaconState.ts

```diff
@@ -17,6 +17,22 @@ import {downloadOrLoadFile} from "../../util";
 import {IBeaconArgs} from "./options";
 import {defaultNetwork, IGlobalArgs} from "../../options/globalOptions";
 import {getGenesisFileUrl} from "../../networks";
+import {WeakSubjectivityServers, getWeakSubjectivityState} from "../weakSubjectivityState";
+import {Checkpoint} from "@chainsafe/lodestar-types/phase0";
+import {SLOTS_PER_EPOCH} from "@chainsafe/lodestar-params";
+
+async function initAndVerifyWeakSujectivityState(
+  config: IBeaconConfig,
+  db: IBeaconDb,
+  wsState: TreeBacked<allForks.BeaconState>,
+  logger: ILogger,
+  wsCheckpoint?: Checkpoint
+): Promise<TreeBacked<allForks.BeaconState>> {
+  if (!isWithinWeakSubjectivityPeriod(config, wsState.genesisTime, wsState, wsCheckpoint)) {
```

Comment:
This check should use the latest stored state's genesisTime, if it exists.

Reply:
I don't think there would be a case where we would have a stored state since we are only getting the weak subjectivity state in the first place if we don't have a state already stored.